egs2/TEMPLATE/st2/st2.sh

```diff
@@ -0,0 +1,2023 @@
+                            # If true, feature extraction and kmeans clustering on the fly
+
+# Tokenization related
+oov="<unk>"         # Out of vocabulary symbol.
+blank="<blank>"     # CTC blank symbol
+sos_eos="<sos/eos>" # sos and eos symbole
+not_avail="<na>"    # not available symbole
+token_joint=false       # whether to use a single bpe system for both source and target languages
+src_token_type=bpe      # Tokenization type (char or bpe) for source languages.
+src_nbpe=30             # The number of BPE vocabulary for source language.
+src_bpemode=unigram     # Mode of BPE for source language (unigram or bpe).
+src_bpe_input_sentence_size=100000000 # Size of input sentence for BPE for source language.
+src_bpe_nlsyms=         # non-linguistic symbols list, separated by a comma, for BPE of source language
+src_bpe_char_cover=1.0  # character coverage when modeling BPE for source language
+tgt_token_type=bpe      # Tokenization type (char or bpe) for target language.
+tgt_nbpe=30             # The number of BPE vocabulary for target language.
+tgt_bpemode=unigram     # Mode of BPE (unigram or bpe) for target language.
+tgt_bpe_input_sentence_size=100000000 # Size of input sentence for BPE for target language.
+tgt_bpe_nlsyms=         # non-linguistic symbols list, separated by a comma, for BPE for target language.
+tgt_bpe_char_cover=1.0  # character coverage when modeling BPE for target language.
+
+# Ngram model related
+use_ngram=false
+ngram_exp=
+ngram_num=3
+
+# Language model related
+use_lm=true       # Use language model for ST decoding.
+lm_tag=           # Suffix to the result dir for language model training.
+lm_exp=           # Specify the directory path for LM experiment.
+                  # If this option is specified, lm_tag is ignored.
+lm_stats_dir=     # Specify the directory path for LM statistics.
+lm_config=        # Config for language model training.
+lm_args=          # Arguments for language model training, e.g., "--max_epoch 10".
+                  # Note that it will overwrite args in lm config.
+use_word_lm=false # Whether to use word language model.
+num_splits_lm=1   # Number of splitting for lm corpus.
+# shellcheck disable=SC2034
+word_vocab_size=10000 # Size of word vocabulary.
+
+# ST model related
+st_tag=        # Suffix to the result dir for st model training.
+st_exp=        # Specify the directory path for ST experiment.
+                # If this option is specified, st_tag is ignored.
+st_stats_dir=  # Specify the directory path for ST statistics.
+st_config=     # Config for st model training.
+st_args=       # Arguments for st model training, e.g., "--max_epoch 10".
+                # Note that it will overwrite args in st config.
+ignore_init_mismatch=false      # Ignore initial mismatch
+num_splits_st=1                # Number of splitting for lm corpus.
+src_case="ts"
+src_lang=wavlm_large_21_km2000  # source language abbrev. id (e.g., wavlm_large_21_km2000)
```

Comment:
As for ST, we use source language transcript to conduct multi-task (without it, training from scratch would be very difficult). Do you have a reason of change the src_lang, instead of adding arguments for discrete processing?

Reply:
This is a good point. Now the names of speech tokens / src text language / src lang / tgt lang are a bit confused. Let me think of some new names.

Reply:
BTW, now it is actually multi-task learning (asr + st, mt is optional). It is okay to train from scratch.

Reply:
Renamed as `speech_token_case/lang` and `src_tgt_text_case/lang`.